include/ddc/kernels/splines/splines_linear_problem_band.hpp

```diff
@@ -155,26 +174,121 @@ class SplinesLinearProblemBand : public SplinesLinearProblem<ExecSpace>
-                        0), // m_q.stride(0) if LAPACK_ROW_MAJOR, m_q.stride(1) if LAPACK_COL_MAJOR
-                m_ipiv.data(),
-                b_host.data(),
-                b_host.stride(0));
-        if (info != 0) {
-            throw std::runtime_error(
-                    "LAPACKE_dgbtrs failed with error code " + std::to_string(info));
+        auto a_device = this->m_a.d_view;
+        auto ipiv_device = this->m_ipiv.d_view;
+
+        std::string name = "gbtrs";
+        Kokkos::RangePolicy<ExecSpace> policy(0, b.extent(1));
+        Kokkos::Profiling::pushRegion(name);
+        if (transpose) {
+            Kokkos::parallel_for(
+                    name,
+                    policy,
+                    KOKKOS_CLASS_LAMBDA(const int i) {
+                        auto sub_b = Kokkos::subview(b, Kokkos::ALL, i);
+                        KokkosBatched::SerialGbtrs<
+                                KokkosBatched::Trans::Transpose,
+                                KokkosBatched::Algo::Gbtrs::Unblocked>::
+                                invoke(a_device, sub_b, ipiv_device, m_kl, m_ku);
+                    });
+        } else {
+            Kokkos::parallel_for(
+                    name,
+                    policy,
+                    KOKKOS_CLASS_LAMBDA(const int i) {
+                        auto sub_b = Kokkos::subview(b, Kokkos::ALL, i);
+                        KokkosBatched::SerialGbtrs<
+                                KokkosBatched::Trans::NoTranspose,
+                                KokkosBatched::Algo::Gbtrs::Unblocked>::
+                                invoke(a_device, sub_b, ipiv_device, m_kl, m_ku);
+                    });
+        }
+        Kokkos::Profiling::popRegion();
+    }
+
+    void solve(
+            typename AViewType::t_dev top_right_block,
+            typename AViewType::t_dev bottom_left_block,
+            typename AViewType::t_dev bottom_right_block,
+            typename PivViewType::t_dev bottom_right_piv,
+            MultiRHS b,
+            bool const transpose) const override
+    {
+        auto Q = this->m_a.d_view;
+        auto piv = this->m_ipiv.d_view;
+
+        MultiRHS b1 = Kokkos::
+                subview(b, std::pair<std::size_t, std::size_t>(0, this->size()), Kokkos::ALL);
+        MultiRHS b2 = Kokkos::
+                subview(b,
+                        std::pair<std::size_t, std::size_t>(this->size(), b.extent(0)),
+                        Kokkos::ALL);
+
+        std::string name = "gbtrs";
+        Kokkos::RangePolicy<ExecSpace> policy(0, b.extent(1));
+        Kokkos::Profiling::pushRegion(name);
+        if (transpose) {
+            Kokkos::parallel_for(
+                    name,
+                    policy,
+                    KOKKOS_CLASS_LAMBDA(const int i) {
+                        auto sub_b1 = Kokkos::subview(b1, Kokkos::ALL, i);
+                        auto sub_b2 = Kokkos::subview(b2, Kokkos::ALL, i);
+
+                        KokkosBatched::SerialGemm<
```

Comment:
You can use Gemv right ? I guess they have some optimizations compared to Gemm

Reply:
I think so. I guess it is also the case for PR #480

Reply:
I don't think so, in #480 this is matrix-matrix multiplication.

Reply:
You are right. It can be Gemv since we operate on subview. I will fix this